docker-gcloud-pubsub-emulator/Dockerfile

```diff
@@ -12,16 +12,31 @@ WORKDIR /src
 RUN go build
 
 
-FROM google/cloud-sdk:alpine
+FROM alpine:latest
+
+ARG TARGETPLATFORM
 
 COPY --from=builder /usr/bin/wait-for /usr/bin
 COPY --from=builder /src/pubsubc /usr/bin
 COPY run.sh /run.sh
 
 RUN apk add --no-cache --update \
         netcat-openbsd \
-        openjdk8-jre && \
-    gcloud components install beta pubsub-emulator
+        openjdk8-jre \ 
+        curl \
+        python3=~3.9
+RUN mkdir gcloud
+WORKDIR /gcloud
+RUN case ${TARGETPLATFORM} in \
+         "linux/amd64")  gcloud_arch=x86_64  ;; \
+         "linux/arm64")  gcloud_arch=ARM  ;; \
+    esac \
+    && curl -O https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-sdk-378.0.0-linux-${gcloud_arch}.tar.gz \
```

Comment:
So unfortunately specifying the gcloud version here breaks the guarantee of this image (https://github.com/TheKevJames/tools/tree/master/docker-gcloud-pubsub-emulator#gcloud-pubsub-emulator) which is that we want to auto-build a new image whenever the gcloud sdk gets updated (which is currently done via dockerhub triggers -- since `google/cloud-sdk` is the base image, dockerhub can be set to autobuild whenever that gets pushed.

We'll need to think of some way to have the same behaviour now that we're building the sdk manually. Two approaches which I could think might work would be:
* always install [the latest version](https://cloud.google.com/sdk/docs/downloads-interactive) and setup a CI job to run on a, say, daily basis which pushes the `:latest` tag, or
* (preferred, but maybe harder?) move the gcloud sdk version tag to some location where [Renovate](https://www.whitesourcesoftware.com/free-developer-tools/renovate/) (which is already enabled for this repo) can update it, then rely on the existing CI jobs / docker pushes to handle things.